### DISCUSSION TITLE
Integrating a Generalised Database Schema with Budibase Forms for CRUD Operations

### DISCUSSION THREAD

**Discussion by Marcello1099** [Budibase/budibase#12142] [Help] [answered]
Hello,

I have devised a generalised database schema to capture various job requirements, and I am keen to adapt this for Budibase Forms, ensuring full CRUD (Create, Read, Update, Delete) capabilities. I shall provide a brief overview of the schema:

### Tables:

**1. `job_requirement_categories`:**

This table captures the distinct categories of job requirements.

| `category_id` (Unique Identifier) | `category_name` (Category Description) |
|----------------------------------|----------------------------------------|
| 1                                | Education Level                        |
| 2                                | Language                               |

**2. `job_requirement_options`:**

This table lists all possible values for each category.

| `option_id` (Unique Identifier) | `category_id` (Linked Category) | `option_name` (Option Description) |
|--------------------------------|--------------------------------|-----------------------------------|
| 1                              | 1                              | Bachelor's Degree                 |
| 2                              | 2                              | English                           |

**3. `job_requirement_values`:**

This table links specific job identifiers to the options in the `job_requirement_options` table.

| `job_fk` (Job Identifier) | `option_id` (Linked Option) |
|--------------------------|-----------------------------|
| JOB123                   | 1                           |
| JOB123                   | 2                           |

### Inquiry:

I would like guidance on setting up Budibase Forms to effectively manage entries based on the above schema, especially given its generalised nature. Are there best practices or recommendations to ensure seamless integration and operation within Budibase? Your insights would be invaluable.

Thank you for your time.

**Comment by melohagan** [comment, marked as answer]
Hey @Marcello1099 

Some suggested reading:

- https://docs.budibase.com/docs/sql-datasource
- https://docs.budibase.com/docs/cascading-dropdown-filters
- https://docs.budibase.com/docs/chaining-actions#example-creating-and-linking-two-table-entries-in-one-form
